mod/beacon/validator/block_builder.go

```diff
@@ -295,7 +295,7 @@ func (s *Service[
 
 	// Dequeue deposits from the state.
 	deposits, err := s.sb.DepositStore().GetDepositsByIndex(
-		depositIndex,
+		depositIndex+1, // pick expected next deposit index
```

Comment:
this come from properly validating and fixing `GetEth1DepositIndex`. Now `GetEth1DepositIndex` returns the index of the latest stored deposit